Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.hxx

```diff
@@ -0,0 +1,170 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+/*=========================================================================
+ *
+ *  Portions of this file are subject to the VTK Toolkit Version 3 copyright.
+ *
+ *  Copyright (c) Ken Martin, Will Schroeder, Bill Lorensen
+ *
+ *  For complete copyright, license and disclaimer of warranty information
+ *  please refer to the NOTICE file at the top of the ITK source tree.
+ *
+ *=========================================================================*/
+
+
+#ifndef itkTriangleMeshCurvatureCalculator_hxx
+#define itkTriangleMeshCurvatureCalculator_hxx
+
+#include <memory> // For unique_ptr
+#include "itkTriangleMeshCurvatureCalculator.h"
+#include "itkObjectFactory.h"
+#include "itkMath.h"
+#include "vnl/vnl_cross.h"
+#include "vnl/vnl_math.h"
+
+namespace itk
+{
+
+template <typename TInputMesh>
+void
+TriangleMeshCurvatureCalculator<TInputMesh>::PrintSelf(std::ostream & os, Indent indent) const
+{
+  this->Superclass::PrintSelf(os, indent);
+  os << indent << "CurvatureType: " << m_CurvatureType << std::endl;
+  itkPrintSelfObjectMacro(TriangleMesh);
+  itkPrintSelfObjectMacro(GaussCurvatureData);
+}
+
+template <typename TInputMesh>
+void
+TriangleMeshCurvatureCalculator<TInputMesh>::Compute()
+{
+  MeshConstPointer inputMesh = this->m_TriangleMesh;
+  if (inputMesh == nullptr)
+  {
+    itkExceptionMacro("First set the Input Triangle Mesh to perform computation");
+  }
+
+  if (this->GetCurvatureType() == TriangleMeshCurvatureCalculatorEnums::Curvatures::GaussCurvature)
+  {
+    this->GetGaussCurvature(inputMesh);
+  }
+  else
+  {
+    itkExceptionMacro("Only Gauss Curvature type available");
+  }
+}
+
+template <typename TInputMesh>
+void
+TriangleMeshCurvatureCalculator<TInputMesh>::GetGaussCurvature(MeshConstPointer inputMesh)
+{
+  if (inputMesh->GetNumberOfCells())
+  {
+    this->ComputeGaussCurvature(inputMesh);
+  }
+}
+
+
+template <typename TInputMesh>
+void
+TriangleMeshCurvatureCalculator<TInputMesh>::ComputeGaussCurvature(MeshConstPointer inputMesh)
+{
+  MeshPointType e0, e1, e2;
+  double        A, alpha0, alpha1, alpha2;
+
+  const unsigned int Nv = inputMesh->GetNumberOfPoints();
+
+  const std::unique_ptr<double[]> K(new double[Nv]);
+  const std::unique_ptr<double[]> dA(new double[Nv]);
+  double                          pi2 = 2.0 * itk::Math::pi;
+  for (unsigned int k = 0; k < Nv; ++k)
+  {
+    K[k] = pi2;
+    dA[k] = 0.0;
+  }
+
+  CellsContainerConstPointer  outCells = inputMesh->GetCells();
+  CellsContainerConstIterator cellsItr = outCells->Begin();
+
+  while (cellsItr != outCells->End())
+  {
+    CellType *               cellPointer = cellsItr.Value();
+    MeshPointIdConstIterator point_ids = cellPointer->GetPointIds();
+
+    MeshPointType v0 = inputMesh->GetPoint(point_ids[0]);
```

Comment:
I am getting compiler warnings for uninitialized variable might be used. This should not occur but should we initialize v0, v1, v2 with some default value to avoid such warnings ?

Reply:
Rather than to v0, v1, v2 I guess it applies to the data/point ids that the cell pointer is returning. Usually `PrintSelf` is helpful at revealing these issues. In either case, we should not have uninitialized variables.

Reply:
v0.Fill(0);
This solves it. I will raise a PR.

Reply:
Note that some errors still persist across the dashboard builds despite Hans' PR had been merged.

Reply:
At line 98 and 132 in itkTriangleMeshCurvatureCalculatorTest.cxx the replacement is remaining for GetGaussCurvatureData to GetModifiableGaussCurvatureData.

Reply:
https://github.com/InsightSoftwareConsortium/ITK/pull/2902